docs/adr/0033-content-server-upgrade-to-react.md

```diff
@@ -0,0 +1,128 @@
+
+## Decision Drivers
+
+- We want to make the changes as safe as possible, minimizing any chance of disruption to the user experience
+- We want to avoid any unnecessary work and clip out extra steps wherever possible.
+- Rolling out changes (and cleaning up upon successful rollout) should be as simple as possible.
+- For the sake of both simplicity and efficiency, we should lean on the decisions made in the `fxa-settings` conversion as much as possible.
+
+## Decisions and Considered Options
+
+### Location of Converted Views
+
+- **Option A: New React App**: the new content-server routes could be hosted in a separate React app alongside the `fxa-settings` app. This would allow us to leave the Settings app untouched and unchanged.
+- **Option B: Shared React App**: we could convert the existing React.js app (currently only used for `fxa-settings`) to also house the new routes.
+- **Option C: Refactor Backbone Views in React**: embed React components in Backbone views, see more details from [a previous ADR here](https://github.com/mozilla/fxa/blob/main/docs/adr/0011-create-new-react-app-for-settings-redesign.md#option-b---take-the-1000-approach-by-creating-all-new-components-with-react-but-modify-existing-backbonemustache-files-or-refactor-these-components-as-react-components).
+
+### Incremental Rollout Approach
+
+#### Context
+
+##### Groups of work
+
+Many routes need to be released in a block or batch, like the routes for the `reset_password` flow, for incremental rollout and to ensure routes that depend on each other, e.g. flows, work as expected. At the time of writing planning is still underway, but there may be as many groups of work as 9, which we can use experiment groups or feature flags we can turn on or off with boolean-valued environment variables to enable.
+
+##### `/beta` URL
+
+In the options below, the `/beta` route refers to an approach we originally took with the new Settings React application where our express framework served Settings at routes that began with `/beta` so "new settings" could be tested at `/beta/settings/*` while "current settings" was still available at `/settings/*`. The alternative is to create a delta at the same path that Express can account for, e.g. `/reset_password` could show the existing content-server route while `/reset_password?showReactApp=true` shows the React version. We don't have access to experiment status at the routing level since our experiment infra spins up at Backbone app start, but we can check for this parameter and if a feature flag is turned on on.
+
+`Pros and cons`: A `/beta` route offers faster development setup but would require a plan of attack to migrate the routes without "beta" at a later time, implement per route, and re-QA, which was not the case for Settings because all routes were under the parent `/settings` route.
```

Comment:
I _think_ I understand what this is saying but I had a very hard time parsing the sentence.

Reply:
Pushed an update, let me know if it needs to be even clearer.